Adds temperature sensors

MrWyss opened the pull request:
Adds temperature sensor support. The entity is linked to the RT-Sensor device. 
![image](https://github.com/user-attachments/assets/6e50edbf-83e6-4980-9c8b-c4992cf1282c)

I couldn't get websocket updates to work. Maybe @mpbzh you can pick up from here.

mpbzh commented:
You're already close :)

What happens, if the websocket receives a new message:
1. `Coordinator.ws_update_data` is called
2. In case of the loads, it calls `self.ws_update_load(data["load"])`
3. This [updates the coordinator's internal state dict](https://github.com/MrWyss/ha-wiser-by-feller/blob/add_temperature_sensors/custom_components/wiser_by_feller/coordinator.py#L257) and notifies its listeners about the update (via `async_set_updated_data` which is implemented in the base class provided by Home Assistant).
4. This in turn notifies all the Entities attached to the coordinator, that they should fetch their updated state. This is implemented [in the Entity base class](https://github.com/MrWyss/ha-wiser-by-feller/blob/add_temperature_sensors/custom_components/wiser_by_feller/entity.py#L101-L105).

This means that we likely don't need separate methods for the different device types (`ws_update_*`), but can move the logic back into the `ws_update_data` method and simplify [the if/else](https://github.com/MrWyss/ha-wiser-by-feller/blob/add_temperature_sensors/custom_components/wiser_by_feller/coordinator.py#L244-L251) to just determine the correct key (`load`, `hvacgroup`, ...)

I see you already have implemented the listener in your `WiserTemperatureSensor`. So you probably only need to remove it to fall back to the WiserEntity's native implementation and make sure the `ws_update_data` updates the state in the the coordinator.

Let me know if that helps!

MrWyss commented:
Thanks for all the input here and on the other channel. I abstracted the classes `WiserEntity → WiserSensorEntity → WiserTemperatureSensorEntity`, simplified `ws_update_data`, and removed `ws_update_*` methods and after understanding 🤔 how the `_states` worked the websocket update works as well 🎉

mpbzh commented on custom_components/wiser_by_feller/coordinator.py:
```diff
@@ -226,41 +236,28 @@ def ws_init(self) -> None:
 
     def ws_update_data(self, data: dict) -> None:
         """Process websocket data update."""
+        _LOGGER.debug("data update received", extra={"data": data})
```
Let's phrase it `Websocket data update received`.

mpbzh commented on custom_components/wiser_by_feller/manifest.json:
```diff
@@ -31,4 +31,4 @@
     }
   ]
-}
+}
```
Missing newline.

MrWyss replied:
🤦

MrWyss replied:
it was some vscode setting that removed it automatically, `"files.insertFinalNewline": true` should fix this from now on.

mpbzh commented on custom_components/wiser_by_feller/coordinator.py:
```diff
@@ -226,41 +236,28 @@ def ws_init(self) -> None:
+            self._states[data["sensor"]["id"]] = data["sensor"]
+        elif "hvacgroup" in data:
+            # TODO: Implement HVAC support issue #7 https://github.com/Syonix/ha-wiser-by-feller/issues/7
+            # Example data: {'hvacgroup': {'id': 87, 'state': {'on': True, 'flags': {...}, 'boost_temperature': 0, 'heating_cooling_level': 0, 'unit': 'C', 'ambient_temperature': 24.1, 'target_temperature': 19.5}}]
```
Here we can already prepare `self._states[data["hvacgroup"]["id"]] = data["hvacgroup"]["state"]`.

Have you double checked whether IDs are unique across device types? Otherwise we would need something like this (untested!):

```python
@dataclass
def WiserState():
    loads: dict[str, dict] = {}
    sensors: dict[str, float | bool] = {}
    hvacgroups: dict[str, dict] = {}
    westgroups: dict[str, dict] = {}

# ...
self._states = WiserStates()
# ...
self._states.loads[<id>] = data
```

MrWyss replied:
I was wondering the same thing, but they seem to be unique, at least on my gateway.

mpbzh commented on custom_components/wiser_by_feller/coordinator.py:
```diff
@@ -226,41 +236,28 @@ def ws_init(self) -> None:
         elif "sensor" in data:
-            self.ws_update_sensor(data["sensor"])
+            # Example data: {'sensor': {'id': 55, 'value': 23.3}}
+            self._states[data["sensor"]["id"]] = data["sensor"]
```
Now that the implementation is done, you can remove the `Example data: ...` comment line.

mpbzh commented on custom_components/wiser_by_feller/coordinator.py:
```diff
@@ -100,6 +102,11 @@ def scenes(self) -> list[Scene] | None:
 
+    @property
+    def sensors(self) -> list[Sensor] | None:
+        """A list of scenes configured in the Wiser by Feller ecosystem (Wiser eSetup app or Wiser Home app)."""
```
"A list of scenes" ;)

mpbzh commented on custom_components/wiser_by_feller/sensor.py:
```diff
@@ -77,3 +91,46 @@ def _handle_coordinator_update(self) -> None:
+
+
+class WiserTemperatureSensorEntity(WiserSensorEntity):
+    """A Wiser temperature sensor (RT-Sensor) entity."""
```
I guess we can just call it a `Wiser room temperature sensor entity`.
